src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -3032,6 +3032,37 @@ internal static void SetUpperUnsafe<T>(in this Vector128<T> vector, Vector64<T>
             Unsafe.AsRef(in vector._upper) = value;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CompExactlyDependsOn(typeof(AdvSimd.Arm64))]
+        internal static int CountMatches<T>(this Vector128<T> vector)
+        {
+            if (AdvSimd.Arm64.IsSupported)
+            {
+                return AdvSimd.Arm64
+                    .AddAcross(AdvSimd.PopCount(vector.AsByte()))
+                    .ToScalar() / (8 * Unsafe.SizeOf<T>());
+            }
+
+            return BitOperations.PopCount(vector.ExtractMostSignificantBits());
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CompExactlyDependsOn(typeof(AdvSimd.Arm64))]
```

Comment:
You don't need CompExactlyDependsOn here, because you already check IsSupported explicitly

Reply:
Ah, I see, thanks.